eve-autonomy/proj_launch
# feat: add launch for status_display_manager

## Description
* Activating status_display_manager node in launch file
* Update README.md for status_display_manager

## Related links
* https://tier4.atlassian.net/browse/AEAP-989
* https://github.com/tier4/dio_ros_driver/pull/5
* https://github.com/eve-autonomy/status_display_manager/pull/1

## Tests performed
* Steps for building the testing packages:
  * git clone `tier4/autoware.proj.eva.git` and checkout the branch `beta/v4.4.0`
  * `mkdir src` and `vcs import src < autoware.repos`
  * git checkout `dio_ros_driver` to [feat/feat/dio_port_array_setting](https://github.com/kodamays/dio_ros_driver) in `src/x1/vendor`
  * git clone `status_display_manager` and checkout to [main](https://github.com/eve-autonomy/status_display_manager) in `src/x1/vendor`
  * git checkout `proj_lauch` to [feat/add_status_display_manager_launch](https://github.com/eve-autonomy/proj_launch/tree/feat/add_status_display_manager_launch) in `src/x1/launcher`
  * Build the package with `colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release --continue-on-error`
* A test was conducted with running Psim (PC simulator) to verify `status_display_manager` is launched correctly with `proj_launch`, when proj_launch is executed. The test ensured that `status_display_manager` is properly initiated as shown as rqt_graph below.
![status_display_manager_rosgraph_2](https://github.com/user-attachments/assets/21e0cfd7-b58d-4eb3-af74-945bcea3e5ec)
* We have confirmed the function worked as stated in [status_display_manager specification](https://github.com/eve-autonomy/status_display_manager?tab=readme-ov-file#overview)
  * [Testing result](https://eva.app.box.com/file/1701005317325) 

## Notes for reviewers


## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
